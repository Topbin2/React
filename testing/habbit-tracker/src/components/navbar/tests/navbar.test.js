import { render, screen } from "@testing-library/react";
import React from "react";
import Navbar from "../navbar";
import renderer from "react-test-renderer";

describe("navbar", () => {
  it("snapshot test", () => {
    const component = renderer.create(<Navbar totalCount="0" />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it("props 에 따라 count의 숫자가 변경되어야 합니다.", () => {
    render(<Navbar totalCount="0" />);
    screen.getByText("0");
  });
});
