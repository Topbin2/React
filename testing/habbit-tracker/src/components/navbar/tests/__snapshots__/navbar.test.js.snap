// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`navbar snapshot test 1`] = `
<nav
  className="navbar"
>
  <i
    className="navbar-logo fas fa-leaf"
  />
  <span>
    Habit Tracker
  </span>
  <span
    className="navbar-count"
    title="count"
  >
    0
  </span>
</nav>
`;
